### PR TITLE
rev1

### PR DESCRIPTION
**Pull request by keremgocen**
- add go app to pull data from firebase periodically
- docker-compose brings up influx and grafana to display metrics
- work in progress

### CODE REVIEW

File: mrmoody-metrics/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM scratch
+CMD ["apt-get install -y ca-certificates"]
+ADD mrmoody /app
+ADD config.json /
+CMD ["./app"]
```
Comment (jacekelgda) [inline mrmoody-metrics/Dockerfile]: pls fix

File: mrmoody-metrics/config.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "msg": "config data",
+    "target_url" : "<firebase-url-here>",
+    "api_key" : "Bearer <your-user-id>",
+    "update_period_minutes" : 2,
+    "db_address" : "http://0.0.0.0:8086",
+    "db_name": "",
+    "db_user": "",
+    "db_pass": ""
+}
```
Comment (jacekelgda) [inline mrmoody-metrics/config.json]: pls fix

File: mrmoody-metrics/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  influx:
+    image: "influxdb:1.1"
+    ports:
+     - "8086"
+  grafana:
+    image: "grafana/grafana"
+    ports:
+     - "3000:3000"
+    links:
+     - influx
```
Comment (jacekelgda) [inline mrmoody-metrics/docker-compose.yml]: pls fix

File: mrmoody-metrics/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+### Metrics for Mr.Moody
+
+### Install Go
+https://golang.org/doc/install
```
Comment (jacekelgda) [inline mrmoody-metrics/README.md]: So, the only way to run it is to install go on host machine ? Or is there an alternate way of running this using docker ?

File: mrmoody-metrics/config.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (jacekelgda) [inline mrmoody-metrics/config.json:1]: if this is a template file, maybe it's better to save it as .dist or .template and the add the actual file to .gitignore ?

File: mrmoody-metrics/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+version: '2'
+services:
+  web:
+    image: "moody"
```
Comment (jacekelgda) [inline mrmoody-metrics/docker-compose.yml]: this is an error - you probably have image moody build locally, that's why it works

File: mrmoody-metrics/main.go [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+func fetchFirebaseData(URL, key string) ([]byte, error) {
+
+	// create request to parse Firebase data
+	req, err := http.NewRequest("GET", URL, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authentication", key)
+
+	// now := time.Now()
```
Comment (jacekelgda) [inline mrmoody-metrics/main.go]: please remove all unused code

File: mrmoody-metrics/main.go [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+	// nowStr := strconv.FormatInt(unixMilli(now), 10)
+	// log.Println("now:", now, " ts:", nowStr)
+	// lastMonth := now.AddDate(0, -3, 0)
+	// lastMonthStr := "1479495795942" //strconv.FormatInt(unixMilli(lastMonth), 10)
+	// log.Println("last month:", lastMonth, " ts:", lastMonthStr)
+
+	q := req.URL.Query()
+	// q.Add("orderBy", "\"date\"")
```
Comment (jacekelgda) [inline mrmoody-metrics/main.go]: please remove all unused code

File: mrmoody-metrics/storage/vault101.go [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+type Votes map[string]Campaign
+
+// Setup creates a new InfluxDB client.
+func Setup(user, pass, addr string) client.Client {
+
+	// NOTE: this assumes you've setup a user and have setup shell env variables,
+	// namely INFLUX_USER/INFLUX_PWD. If not just omit Username/Password below.
+	c, err := client.NewHTTPClient(client.HTTPConfig{
+		Addr: addr,
+		// Username: os.Getenv("INFLUX_USER"),
```
Comment (jacekelgda) [inline mrmoody-metrics/storage/vault101.go]: please remove all unused code

**Comment by keremgocen** [comment]
Closing this to make a new PR with Docker setup and docs only.

**Pull request closed without merging**